### PR TITLE
Fix tagged enum Bigint bug

### PR DESCRIPTION
**Pull request by mfish33**
Closes: #30 

Sorry for the late reply. I have gotten very busy with school in the last week. This should fix the tagged enum issue by adding an option for big_ints in the deserialize any function. I have also added a test case to match. Do you have any suggestions about the comment regarding u64's?

### CODE REVIEW

**Comment by mfish33** [comment]
Also sorry for the bad git log. It will be fixed if I do another PR

**Comment by RReverser** [comment]
> Also sorry for the bad git log. It will be fixed if I do another PR

I think you can just do `git rebase`.

**Comment by mfish33** [comment]
> I think you can just do git rebase.
I tried but I am having a hard time due to the merges. I can either create a new PR or they will just be squashed when this PR gets merged

**Comment by RReverser** [comment]
> I tried but I am having a hard time due to the merges.

There shouldn't be any conflicts. Just do interactive rebase and delete all the commits that are already on master.

File: src/de.rs [hunk 1]
```diff
@@ -246,6 +246,9 @@ impl<'de> de::Deserializer<'de> for Deserializer {
             visitor.visit_unit()
         } else if let Some(v) = self.value.as_bool() {
             visitor.visit_bool(v)
+        } else if self.value.is_bigint() {
+            // TODO: Find a way to properly support tagged enums deserializing numbers between i64::MAX and u64::MAX
```
Comment (RReverser) [inline src/de.rs:250]: I'm not sure I understand this comment. Are you saying there are values for which it wouldn't work right now?

Comment (mfish33) [reply src/de.rs]: I ended up fixing this by trying both i64 and u64 and then erroring if it can not deserialize into either.

File: src/de.rs [hunk 2]
```diff
@@ -233,6 +233,26 @@ impl Deserializer {
             _ => self.invalid_type(visitor),
         }
     }
+
+    fn bigint_to_i64(&self, bigint: &BigInt) -> Option<i64> {
```
Comment (RReverser) [inline src/de.rs:237]: Those don't use `self`, so I'd suggest extracting them into freestanding functions before `impl Deserializer`.

Comment (mfish33) [reply src/de.rs]: Good call

**Review by RReverser — APPROVED** [review]

File: src/de.rs (hunk 1 above)
Comment (RReverser) [reply src/de.rs]: Sounds reasonable.

**Comment by RReverser** [comment]
Fix released in 0.4.3.

**Pull request merged**